src/commands/deleteVirtualMachine/getResourcesAssociatedToVm.ts

```diff
@@ -0,0 +1,90 @@
+/*---------------------------------------------------------------------------------------------
+ *  Copyright (c) Microsoft Corporation. All rights reserved.
+ *  Licensed under the MIT License. See License.txt in the project root for license information.
+ *--------------------------------------------------------------------------------------------*/
+
+import { ComputeManagementClient } from "@azure/arm-compute";
+import { NetworkManagementClient, NetworkManagementModels } from "@azure/arm-network";
+import { localize } from "../../localize";
+import { VirtualMachineTreeItem } from "../../tree/VirtualMachineTreeItem";
+import { createComputeClient, createNetworkClient } from "../../utils/azureClients";
+import { getNameFromId } from "../../utils/azureUtils";
+import { networkInterfaceLabel, ResourceToDelete, virtualNetworkLabel } from "./deleteConstants";
+
+export async function getResourcesAssociatedToVm(node: VirtualMachineTreeItem): Promise<ResourceToDelete[]> {
+
+    const associatedResources: ResourceToDelete[] = [];
+
+    const networkNames: string[] = [];
+    if (node.data.networkProfile?.networkInterfaces) {
+        for (const networkRef of node.data.networkProfile?.networkInterfaces) {
+            if (networkRef.id) {
+                networkNames.push(getNameFromId(networkRef.id));
+            }
+        }
+
+    }
+
+    const resourceGroupName: string = node.resourceGroup;
+    const networkClient: NetworkManagementClient = await createNetworkClient(node.root);
+    for (const networkName of networkNames) {
+        // if we fail to get a resource, we keep trying to get all associated resources we can rather than erroring out
+        try {
+            const networkInterface: NetworkManagementModels.NetworkInterface = await networkClient.networkInterfaces.get(resourceGroupName, networkName);
+            associatedResources.push({
+                resourceName: networkName, resourceType: networkInterfaceLabel,
+                deleteMethod: async (): Promise<void> => { await networkClient.networkInterfaces.deleteMethod(resourceGroupName, networkName); }
+            });
+
+            if (networkInterface.ipConfigurations) {
+                for (const ipConfigurations of networkInterface.ipConfigurations) {
+                    if (ipConfigurations.publicIPAddress?.id) {
+                        const publicIpName: string = getNameFromId(ipConfigurations.publicIPAddress.id);
+                        associatedResources.push({
+                            resourceName: publicIpName, resourceType: localize('publicip', 'public IP address'),
+                            deleteMethod: async (): Promise<void> => { await networkClient.publicIPAddresses.deleteMethod(resourceGroupName, publicIpName); }
```

Comment:
Rather than use "resourceGroupName" for all resources, I feel like we want to do something like this:

```typescript
const publicIpRg: string = getResourceGroupFromId(ipConfigurations.publicIPAddress.id);
```

I'm not entirely sure how we want to handle associated resources in a different group, but I think we need to do something